source/validate_cfg.cpp

```diff
@@ -424,14 +424,63 @@ spv_result_t PerformCfgChecks(ValidationState_t& _) {
       for (auto edge : postdom_edges) {
         edge.first->SetImmediatePostDominator(edge.second);
       }
-      /// calculate back edges.
+      /// Calculate back edges.
       DepthFirstTraversal(
           function.pseudo_entry_block(),
           function
               .AugmentedCFGSuccessorsFunctionIncludingHeaderToContinueEdge(),
           ignore_block, ignore_block, [&](cbb_ptr from, cbb_ptr to) {
             back_edges.emplace_back(from->id(), to->id());
           });
+
+      // Perform a DepthFirstTraversal that ensures CFG nesting depth is valid.
+      const int cfg_depth_limit = 1023;
+      bool cfg_depth_limit_violated = false;
+      int depth = 0;
+      std::unordered_map<cbb_ptr, int> block_depth;
+      DepthFirstTraversal(
+          function.first_block(), function.AugmentedCFGSuccessorsFunction(),
+          [&](cbb_ptr b) {
```

Comment:
The style guide tells us to avoid overly-broad lambda captures.  In this case I think you only need four new variables you added here.

(Yes, I see there are other cases in this code. I was cutting some slack earlier.)
The situation would be better if you pulled this section of functionality into its own function. Then there's less surrounding state to capture in the first place.